pkg/server/server.go

```diff
@@ -102,3 +114,18 @@ func New(configuration options.Configuration) (controllerruntime.Manager, error)
 
 	return mgr, nil
 }
+
+func k8sCommunicationCheck(client *discovery.DiscoveryClient) (err error) {
```

Comment:
query: Wondering if there is a readiness probe that can be set against the deployment yaml specs to handle just this? We may not need to code (however this puts the burden on the end users)

Reply:
The scenario I am trying to prevent is when metacontroller comes up it is much faster than the istio sidecar so metacontroller will start out in a crashloopbackoff when it attempts to communicate with the k8s api server since the sidecar proxy is not available. I thought about the readiness probe as well, but I believe that will only prevent routing to the service and not the underlying service from executing calls out to k8s.

Side note: Istio does have an annotation that could achieve this https://istio.io/latest/news/releases/1.7.x/announcing-1.7/change-notes/#traffic-management. However, we will be migrating to a new service mesh that does not have this feature so this PR is an attempt to make it more generic.

Reply:
@mjsmith1028 Thanks for explaining. 
Now wondering if it also makes sense to implement liveness & readiness endpoints that are builtin inside metacontroller. Perhaps in some other PRs. cc @grzesuav

Reply:
I was thinking of liveness/readiness, however unsure how it can be checked in other way that just returning `ok` from server. Will think and see if controller runtime have sth build in .